crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -140,6 +141,75 @@ impl Settlement {
 
         Ok(Self { trades, auction_id })
     }
+
+    /// Build a settlement from a `competition::Solution`.
+    ///
+    /// Since `competition::Solution` does not contain JIT orders, the resulting
+    /// settlement might be a subset of it's onchain observable twin.
+    pub fn from_solution(
+        solution: &competition::Solution,
+        auction: &auction::Auction,
+        auction_id: auction::Id,
+    ) -> Result<Self, error::Solution> {
+        let auction_orders = auction
+            .orders
+            .iter()
+            .map(|o| (o.uid, o))
+            .collect::<HashMap<_, _>>();
+
+        let mut trades = Vec::with_capacity(solution.orders().len());
+        for (order_uid, traded) in solution.orders().iter() {
+            let order = auction_orders
+                .get(order_uid)
+                .ok_or(error::Solution::MissingOrder)?;
+            let sell_token = order.sell_token.into();
+            let buy_token = order.buy_token.into();
+            trades.push(trade::Trade::new(
+                order.uid,
+                eth::Asset {
+                    token: sell_token,
+                    amount: order.sell_amount.into(),
+                },
+                eth::Asset {
+                    token: buy_token,
+                    amount: order.buy_amount.into(),
+                },
+                order.side,
+                match order.side {
+                    order::Side::Sell => traded.sell.0.into(),
+                    order::Side::Buy => traded.buy.0.into(),
+                },
+                trade::Prices {
+                    uniform: trade::ClearingPrices {
+                        sell: solution
+                            .prices()
+                            .get(&sell_token)
+                            .ok_or(error::Solution::MissingClearingPrice)?
+                            .get()
+                            .0,
+                        buy: solution
+                            .prices()
+                            .get(&buy_token)
+                            .ok_or(error::Solution::MissingClearingPrice)?
+                            .get()
+                            .0,
+                    },
+                    custom: trade::ClearingPrices {
+                        sell: match order.side {
+                            order::Side::Sell => traded.buy.into(),
+                            order::Side::Buy => traded.sell.into(),
+                        },
+                        buy: match order.side {
+                            order::Side::Sell => traded.sell.into(),
+                            order::Side::Buy => traded.buy.into(),
+                        },
```

Comment:
Why is this correct? Shouldn't the sell price always be `traded.sell.into()`?

Reply:
Good catch! Buy orders had a bug. I've fixed it and added a comment why prices are reversed version of traded amounts.